base/math.jl

```diff
@@ -166,6 +170,43 @@ function hypot{T<:AbstractFloat}(x::T, y::T)
     x * sqrt(one(r)+r*r)
 end
 
+# 3-arg hypot
+hypot(x::Real, y::Real, z::Real) =
+    hypot(promote(float(x), float(y), float(z))...)
+function hypot{T<:AbstractFloat}(x::T, y::T, z::T)
+    x = abs(x)
+    y = abs(y)
+    z = abs(z)
+    m = max(x, y, z)
+    if isinf(m)
+        return m
+    elseif isfinite(3*m*m) && m*m != 0 # Scaling not necessary
```

Comment:
This doesn't work if `T` is an integer type. I think the easiest way to handle this would be to explicitly convert to a floating type first, before calling `max`.